hr_timesheet_sheet_no_negative/models/account_analytic_line.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    def _negative_unit_amount(self):
+        if not self.company_id:
+            user_company = self.env.user.company_id
+            company_id = self._context.get('company_id') or user_company.id
+            company = self.env['res.company'].browse(company_id)
+        else:
+            company = self.company_id
+        allow_negative = company.timesheet_negative_unit_amount
+        return allow_negative
+
+    @api.onchange('unit_amount')
+    def onchange_unit_amount(self):
+        self.ensure_one()
+        if not self._negative_unit_amount() and self.project_id \
+                and self.unit_amount < 0.0:
+            self.write({'unit_amount': 0.0})
```

Comment:
Here you prevent the user to enter negative amounts on an analytic line, regardless whether the line belongs to a timesheet sheet or not, but the module is named `hr_timesheet_sheet_no_negative`. :thinking: Anyway, in general, I still have doubts about why this module would be needed.

Reply:
Well, the `and self.project_id` assures only applies to timesheets, not to any analytic line.

Reply:
No, you mentioned "analytic line" before. So I explained that the condition doesn't apply to all analytic lines, only applies to those that have a project, and that means, to those that its `unit_amount` means `hours`.

Reply:
I see that you updated the readme, explaining that it also applies to the timesheets and not only to the timesheets lines.
:+1: